src/app/shared/infinite-scroll/infinite-scroll.directive.ts

```diff
@@ -0,0 +1,35 @@
+import {Directive, EventEmitter, HostListener, Input, Output} from '@angular/core';
+import {debounceTime, Subject} from 'rxjs';
+import {InfiniteScrollDirection} from './infinite-scroll-direction.enum';
+
+@Directive({
+    selector: '[appInfiniteScroll]',
+})
+export class InfiniteScrollDirective {
+    @Input() borderOffset: number;
+    @Input() delay: number;
+    @Output() loadData = new EventEmitter<InfiniteScrollDirection>();
+
+    private readonly scrollEventSubject = new Subject<InfiniteScrollDirection>();
+    @HostListener('scroll', ['$event'])
+    onScroll(event: Event): void {
```

Comment:
```suggestion
    @HostListener('scroll', ['$event.target'])
    onScroll(target: HTMLElement): void {
```

Или я бы порекомендовал так:
```ts
    @HostListener('scroll', ['$event.target'])
    onScroll({scrollTop, scrollHeight, clientHeight}: HTMLElement): void {
```